src/core/graph.ts

```diff
@@ -296,14 +276,57 @@ export class DependencyGraph<T> {
     return [this.adjacencyMap.size, this.numberOfEdges];
   }
 
-  nodes(): IterableIterator<T> {
+  getNodes(): IterableIterator<T> {
     return this.adjacencyMap.keys();
   }
 
+  toString: () => string = () => this.toMermaidRepresentation();
+
+  /**
+   * Return a representation that conforms with the syntax of mermaid.js
+   * @param edgesWithIssue Edges in the **dependency** graph that causes issues
+   * to the execution. A set containing arrays with [effect, origin].
+   */
+  toMermaidRepresentation(edgesWithIssue?: Set<[T, T]>): string {
```

Comment:
`edgesWithIssue` is for situations that might arise here: 

https://github.com/lf-lang/reactor-ts/blob/c01a31c7d400bd5a31e9707e4293cfa791d886b7/src/core/reactor.ts#L1130-L1145

If the introduction of edge(s) makes the graph problematic, such as introducing `app.R[R2] -> app.R[R5]` in this graph (this is the precedence graph; in reality a dependency, i.e. edge, from R5 to R2 is introduced):

```mermaid
graph
    0["app.R[R3]"]
    1["app.R[R5]"]
    2["app.R[R4]"]
    3["app.R[R2]"]
    4["app.R[R0]"]
    5["app.R[R6]"]
    1 --> 0
    0 --> 3
    5 --> 3
    2 --> 4
```

Then it will become this indicating that the newly-added edge is problematic.
```mermaid
graph
    0["app.R[R3]"]
    1["app.R[R5]"]
    2["app.R[R4]"]
    3["app.R[R2]"]
    4["app.R[R0]"]
    5["app.R[R6]"]
    1 --> 0
    0 --> 3
    5 --> 3
    2 --> 4
    3 --x 1
```

Reply:
Using Mermaid is such an improvement!